src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -1197,6 +1122,10 @@ class CVulkanLogicalDevice final : public ILogicalDevice
     }
             
 private:
+    core::smart_refctd_ptr<IDriverMemoryAllocation> allocateGPUMemory(
+        const IDriverMemoryBacked::SDriverMemoryRequirements& reqs,
+        VkMemoryPropertyFlags desiredMemoryProperties);
```

Comment:
I see the disconnect, there was never a plain simple
```cpp
virtual core::smart_refctd_ptr<IDriverMemoryAllocation> allocateMemory(const IDriverMemoryBacked::SDriverMemoryRequirements& fullReqs) {return nullptr;}
```
to override in `ILogicalDevice`

Reply:
I have an idea about this:

We can get rid of all `allocate**Memory` functions and just keep one simple `allocateGPUMemory` function which will be responsible for deducing all the Vulkan flags. To use it correctly, the user __must__ specify correct flags otherwise it fails.
For common cases, they can make use of those 5 `static inline` functions to get common, predefined memory type settings (which may or may not be supported by the device). It will require minimal changes, for example, the constructor for `IUtilities` will require literally zero changes.

What do you think?

Reply:
`allocateMemory` should fail if user specifies unsupported flags (no patching up or deduction of arguments)

The `static inline` non-overridable functions get you some basic predefined memory type settings.

There should be a separate `virtual bool` function for patching up/deducing the actual parameters to pass to `allocateMemory`
That one should take into account the runtime capabilities of the `ILogicalDevice`, return false if its impossible to patch up (catastrophic failure, i.e. you wanted mappable memory, but no such heap)

Reply:
Obsolete, discussed on discord.